packages/expo-image/android/src/main/java/expo/modules/image/ExpoImageModule.kt

```diff
@@ -42,4 +56,28 @@ class ExpoImageModule(val context: ReactApplicationContext) : ReactContextBaseJa
       }
     }
   }
+
+  @ReactMethod
+  fun getSize(url: String, promise: Promise) {
+    try {
+      Glide
+          .with(context)
+          .`as`(ExpoImageSize::class.java)
+          .apply(sizeOptions)
+          .load(url)
+          .into(object : SimpleTarget<ExpoImageSize>() {
+            private val promiseRef = promise
+
+            override fun onResourceReady(resource: ExpoImageSize, transition: Transition<in ExpoImageSize>?) {
+              promiseRef.resolve(resource.asWritableNativeMap())
+            }
+
+            override fun onLoadFailed(errorDrawable: Drawable?) {
+              promiseRef.reject("ERR_IMAGE_GETSIZE_FAILURE", "Failed to get size of the image: $url")
+            }
+          })
+    } catch (e: Exception) {
+      promise.reject("ERR_IMAGE_GETSIZE_FAILURE", "Failed to get size of the image: $url", e)
```

Comment:
Same here 😉

Reply:
I can improve the message here by printing content of `e.message` -- done in recent commit.